src/emulator.c

```diff
@@ -50,6 +51,10 @@ struct ofono_emulator {
 	guint callsetup_source;
 	int pns_id;
 	struct ofono_handsfree_card *card;
+	int r_codecs;
+	uint8_t negotiated_codec;
+	uint8_t proposed_codec;
+	guint delay_sco;
```

Comment:
To be extra-safe call g_source_remove() for this ID if != 0 when we destroy the emulator.